### PR TITLE
Add blockchain:reset command - Closes #284

### LINKED ISSUE #284: Add core blockchain:reset command
### Description

Add `blockchain:reset` command, according to the following specification:

1. It should remove all the data saved in `blockchain.db`
2. It should support:

```
--data-path (-d / LISK_DATA_PATH)
  Specifies which data path the application should use
--yes (-y)
  Skips the prompt to confirm the deletion.
```
3. It should ask confirmation if not `--yes` is not provided.

### Motivation

It should be able to reset the blockchain data in case the node want to start from scratch.

### Acceptance Criteria

- Add unit tests
- Examples and description are clear and correct
- Check if proper error shows if the data does not exist
- Check if it saves in the current working directory if output is not specified
- Check if it saves to the specified path when the output is specified

### Additional Information

Related issues #254

### PR DESCRIPTION
**Pull request by mitsuaki-u**
### What was the problem?

This PR resolves #284 

### How was it solved?

- Add blockchain:reset command
- Add related tests

### How was it tested?

- `npm run test`
- ./run blockchain:reset
- ./run blockchain:reset --data-path ./my-path
- ./run blockchain:reset --yes

### CODE REVIEW

File: src/commands/blockchain/reset.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+		'blockchain:reset --yes',
+	];
+
+	static flags = {
+		'data-path': flagParser.string({
+			char: 'd',
+			description:
+				'Directory path to specify where node data is stored. Environment variable "LISK_DATA_PATH" can also be used.',
+			env: 'LISK_DATA_PATH',
+		}),
```
Comment (ManuGowda) [inline src/commands/blockchain/reset.ts:36]: 'data-path': flagParser.string({
			...commonFlags.dataPath,
			env: 'LISK_DATA_PATH',
		}),

you can refer start.ts file

File: test/commands/blockchain/reset.spec.ts [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+				.it('should reset the blockchain db', () => {
+					expect(KVStoreStubInstance.clear).to.have.been.calledOnce;
+				});
+		});
+
+		describe('when reset with data-path', () => {
+			setupTest()
+				.stub(appUtils, 'isApplicationRunning', sandbox.stub().returns(false))
+				.command(['blockchain:reset', '--data-path=/my/app/'])
+				.it('should create db object for "blockchain.db" for given data path', () => {
```
Comment (ishantiw) [inline test/commands/blockchain/reset.spec.ts:121]: Does it really create a new db object maybe `Should reset blockchain db at the given data path`?

File: test/commands/blockchain/reset.spec.ts [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+				.it('should reset the blockchain db', () => {
+					expect(KVStoreStubInstance.clear).to.have.been.calledOnce;
+				});
+		});
+
+		describe('when skipping confirmation prompt', () => {
+			setupTest()
+				.stub(appUtils, 'isApplicationRunning', sandbox.stub().returns(false))
+				.command(['blockchain:reset', '--yes'])
+				.it('should create db object for "blockchain.db" for given data path', () => {
```
Comment (ishantiw) [inline test/commands/blockchain/reset.spec.ts:153]: same here, seems like there is no data path given here so maybe just `Should reset blockchain db`

File: test/commands/blockchain/reset.spec.ts (hunk 2 above)
Comment (mitsuaki-u) [reply test/commands/blockchain/reset.spec.ts:121]: This test is checking if `getBlockchainDB` so it does create new KVStore object. There is a specific test for the actual reset below.

**Pull request merged**